### PR TITLE
Improved performance

### PR DESCRIPTION
**Pull request by jsilvestre**
Some small changes to improve performance, part of the work on big lists.

* The first one is event delegation. I took the code from @m4dz work in #226.

* I also find a bug where a high cost operation was performed whereas it should not. Getting a folder's content with 1000 elements is now <1s, whereas it was around 18s before. It doesn't remove the slow feeling in the client, but it's a massive speed increase.

### CODE REVIEW

**Comment by m4dz** [comment]
> The first one is event delegation. I took the code from @m4dz work in #226.

:kissing: !!

If you can, after revamp, use SVG symbols for actions icons too, I'll be your obedient servant :smile:.

File: client/app/views/file_view.coffee [hunk 1]
```diff
@@ -135,18 +111,32 @@ module.exports = class FileView extends BaseView
...
 
-            @listenTo @uploadQueue, 'add remove reset', =>
-                hasItems = @uploadQueue.getNumUploadingElementsByPath(path) > 0
-                if hasItems and not @$('.spinholder').is ':visible'
-                    @showLoading()
-                else if not hasItems and @$('.spinholder').is ':visible'
-                    @hideLoading()
 
-            @listenTo @uploadQueue, 'upload-complete', =>
-                @hasUploadingChildren = false
```
Comment (jsilvestre) [inline client/app/views/file_view.coffee]: Please don't merge, this is wrong, this should not be delegated, or at least not this way.

**Comment by jsilvestre** [comment]
There is one more thing I'd like to do, in order to remove complexity from the "select" mechanism. I screwed it, it could be much easier to understand.

**Comment by jsilvestre** [comment]
I added two changes:

* elements to upload are managed into a separate collection, because using a projection was handy but not efficient at all. Since the upload queue's API is pretty clear, the base collection and the upload queue collection should be kept in sync easily

* there was a bug in event delegation: when an item was removed (the view would not exist anymore, as a result it couldn't call the method).

**Comment by jsilvestre** [comment]
By the way, those changes require intensive testing. I did a lot of them, but my data set might different from someone's else.

**Comment by frankrousseau** [comment]
OK thank you  @jsilvestre

**Pull request merged**